src/subdomains/staking/application/services/liquidity-management.service.ts

```diff
@@ -127,19 +137,30 @@ export class LiquidityManagementService {
       MasternodeState.PRE_RESIGNED,
       BlockchainMasternodeState.RESIGNED,
     );
+    await this.checkIfOwnerIsEmpty(
+      allInProcessMasternodes.filter((node) => node.state === MasternodeState.MOVING_COLLATERAL),
+    );
```

Comment:
Why not use the normal handleMasternodesWithState? We could return undefined (or similar) from txFunc, if update cannot be applied already. BlockchainMasternodeState check could theoretically be moved there too.

Reply:
thought about that and then decided against it. as we would need to make one of the functions async, as we will need to ask for the balance of the owner address. and then update.

IMHO those were too much changes to force it into our current setup and made an additional function instead

Reply:
txFunc and updateFunc are already async. Or which functions did you mean?

Reply:
hm... yeah true. but we missed to write async as keyword before every function. But we are returning a promise. should we add async everywhere?

Reply:
If async or not has no effect on calling methods (you e.g. cannot have abstract async methods). It just permits to use the await keyword in the method and to return normal values instead of promises.

Reply:
So basically we can just just add it where required.